packages/react-magma-dom/src/components/FormFieldContainer/FormFieldContainer.tsx

```diff
@@ -188,12 +204,14 @@ export const FormFieldContainer = React.forwardRef<
           labelWidth={labelWidth}
         >
           {children}
-          {typeof maxLength === 'number' && (
+          {typeof countProps === 'number' && hasCharacterCounter && (
             <CharacterCounter
+              hasCharacterCounter={hasCharacterCounter}
               id={descriptionId}
               inputLength={inputLength}
               isInverse={isInverse}
-              maxLength={maxLength}
+              maxCount={countProps}
+              maxLength={countProps}
```

Comment:
Just want to triple check, should this be as follows?
```suggestion
              maxCount={maxCount}
              maxLength={maxLength}
```

Reply:
Either seems to work, shall I make the amendment?

Reply:
If both work, I think we should update it. It'll be safer and easier to get rid of the props that are temporary when it's time.

Reply:
Actually `countProps` should remain, it breaks and is undefined in the tests otherwise.

Reply:
Here's what I think:

1. In FormFieldContainer, we should pass maxCount to maxCount and maxLength to maxLength. This will ensure that the CharacterCounter component receives the appropriate values.
2. Then, in the CharacterCounter component, we do the extra handling to figure out which value to use. This means using maxCount when that has a number value, or maxLength as the fallback. Something like
```
  // Temporary while we need to support both maxCount and maxLength.
  const maxCharacters = typeof maxCount === 'number' ? maxCount : maxLength;
```
3. Then, in the rest of the CharacterCounter component we use the `maxCharacters` variable to check all the logic. This of course can be called whatever you want.

I think the benefit of doing this is that there's less risk of the values being accidentally interchanged. How it's working right now is that FormFieldContainer is sending both maxCount and maxLength the same value. Technically it's working, but could potentially cause problems. 


There are multiple ways of doing this, this is just a suggestion. What do you think?